### ISSUE TITLE
undesirable_function_linter fails to match non-syntactic names

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1911]
```r
lintr::lint("`%%`(x, y)\n", lintr::undesirable_function_linter(c("%%" = NA)))
lintr::lint("`%%`(x, y)\n", lintr::undesirable_function_linter(c("`%%`" = NA)))
# <text>:1:1: style: [undesirable_function_linter] Function "`%%`" is undesirable. Disable this with a comment: `# nolint` (one line) or `# lintr: disable` (multiple lines). See go/tidy-style#linting for more.
# `%%`(x, y)
# ^~~~
```

It seems should either match in the first case, or perhaps throw a warning/error that non-syntactic names must be wrapped in backticks.

But the latter also seems off -- we apparently can't match a call in quoted form:

```
lintr::lint("`%%`(x, y)\n", lintr::undesirable_function_linter(c('"%%"' = NA)))
lintr::lint('"%%"(x, y)\n', lintr::undesirable_function_linter(c('"%%"' = NA)))
```

(no lints from either call)
